Game engine: implement bidding moves

nockty opened the pull request:
Implement moves for the bidding phase of the game. Add some unit tests.

Commit:
:construction: Implement bidding moves

Commit:
:white_check_mark: Add tests for bidding moves

Scrabouillmax reviewed (approved):
It would be nice to test the comparison as well ;)

Commit:
:white_check_mark: Add ComparisonTestCase helper

Add a new abstract class that can be used for comparisons
and update the tests for contracts to use this new helper.

Commit:
:ok_hand: Add test for comparing bidding moves
